cli/azd/cmd/util.go

```diff
@@ -169,94 +186,77 @@ func ensureEnvironmentInitialized(ctx context.Context, environmentName string, e
+				return fmt.Errorf("prompting for location: %w", err)
+			}
 		}
 
 		env.Values[environment.LocationEnvVarName] = strings.TrimSpace(location)
 	}
 
-	azCli := commands.GetAzCliFromContext(ctx)
-
 	if !hasSubID {
-		subscriptionInfos, err := azCli.ListAccounts(ctx)
-		if err != nil {
-			return fmt.Errorf("listing accounts: %w", err)
-		}
-
-		sort.Sort(azureutil.Subs(subscriptionInfos))
-
-		// If `AZURE_SUBSCRIPTION_ID` is set in the environment, use it to influence
-		// the default option in our prompt. Fall back to the what the `az` CLI is
-		// configured to use if the environment variable is unset.
-		defaultSubscriptionId := os.Getenv(environment.SubscriptionIdEnvVarName)
-		if defaultSubscriptionId == "" {
-			for _, info := range subscriptionInfos {
-				if info.IsDefault {
-					defaultSubscriptionId = info.Id
-				}
-			}
-		}
-
-		var subscriptionOptions = make([]string, len(subscriptionInfos)+1)
-		var defaultSubscription string
-
-		for index, info := range subscriptionInfos {
-			subscriptionOptions[index] = fmt.Sprintf("%2d. %s (%s)", index+1, info.Name, info.Id)
-
-			if info.Id == defaultSubscriptionId {
-				defaultSubscription = subscriptionOptions[index]
-			}
-		}
-
-		subscriptionOptions[len(subscriptionOptions)-1] = "Other (enter manually)"
-
-		var subscriptionId string
+		var subscriptionId string = envSpec.subscription
 
-		for env.GetSubscriptionId() == "" {
-			var subscriptionSelection string
-			err := askOne(&survey.Select{
-				Message: "Please select an Azure Subscription to use:",
-				Options: subscriptionOptions,
-				Default: defaultSubscription,
-			}, &subscriptionSelection)
+		if subscriptionId == "" {
+			subscriptionOptions, defaultSubscription, err := getSubscriptionOptions(ctx)
 			if err != nil {
-				return fmt.Errorf("reading subscription id: %w", err)
+				return err
 			}
-			if subscriptionSelection == "Other (enter manually)" {
-				err = askOne(&survey.Input{
-					Message: "Enter an Azure Subscription to use:",
-				}, &subscriptionId)
+
+			for subscriptionId == "" {
+				var subscriptionSelection string
+				err := askOne(&survey.Select{
+					Message: "Please select an Azure Subscription to use:",
+					Options: subscriptionOptions,
+					Default: defaultSubscription,
+				}, &subscriptionSelection)
 				if err != nil {
 					return fmt.Errorf("reading subscription id: %w", err)
 				}
-			} else {
-				subscriptionId = subscriptionSelection[len(subscriptionSelection)-len("(059cdffa-0e5b-47d8-ad4b-f13fd9099f21)")+1 : len(subscriptionSelection)-1]
+				if subscriptionSelection == manualSubscriptionEntryOption {
+					err = askOne(&survey.Input{
+						Message: "Enter an Azure Subscription to use:",
+					}, &subscriptionId)
+					if err != nil {
+						return fmt.Errorf("reading subscription id: %w", err)
+					}
+				} else {
+					subscriptionId = subscriptionSelection[len(subscriptionSelection)-len("(059cdffa-0e5b-47d8-ad4b-f13fd9099f21)")+1 : len(subscriptionSelection)-1]
```

Comment:
Can somebody explain what this magic GUID is about? I'd like to add a comment about that

Reply:
Changed to "nil UUID" here to make more clear this UUID is just to ensure subscription ID is extracted correctly